### PR TITLE
Fix DM-2708: QueryResource corrupted after log statement

### PR DESCRIPTION
**Pull request by smonkewitz**
QueryResource constructor was stashing a raw pointer to the string
buffer underlying one of its input arguments into a base class
member variable.  This failed unpredictably in the case where the
input argument was a temporary (depending on what code followed.)

Instead now QueryResource ctor copies the input data onto the
heap, then frees the heap allocation in its dtor.

### CODE REVIEW

**Comment by smonkewitz** [comment]
Should you:
```
#include <string.h>
#include <cstdlib>
```
in QueryResource.h to avoid depending on other headers bringing these in?

File: core/modules/qdisp/QueryResource.h [hunk 1]
```diff
@@ -56,17 +56,23 @@ class QueryResource : public XrdSsiService::Resource {
...
+          _status(status)
+    {
+        if (rName == NULL) {
+            throw std::bad_alloc();
+        }
     }
 
-    virtual ~QueryResource() {}
+    virtual ~QueryResource() {
+        free(const_cast<char *>(rName)); // clean up heap allocated resource path copy
```
Comment (smonkewitz) [inline core/modules/qdisp/QueryResource.h]: nitpick: `std::free` or `::free` (depending on whether you included `<cstdlib>` or `<stdlib.h>`)

Comment (fritzm) [inline core/modules/qdisp/QueryResource.h]: wilco

**Comment by fritzm** [comment]
wilco on includes as well, thx

**Pull request merged**